src/warplib/implementations/types/bytes.ts

```diff
@@ -0,0 +1,168 @@
+
+          let value_at_index_felt252 = u128_to_felt252(value_at_index_u128);
+          let value_at_index_u8 = u8_try_from_felt252(value_at_index_felt252).unwrap();
+
+          bytes1{ value: value_at_index_u8}
+        }
+        fn length(self: bytes${length}) -> u8 {
+          ${length}_u8
+        }
+        // Comparison
+        fn less_or_equal_comparison(self: bytes${length}, other: bytes${length}) -> bool {
+          self.value <= other.value
+        } 
+        fn less_comparison(self: bytes${length}, other: bytes${length}) -> bool {
+          self.value < other.value
+        }
+        fn equal_comparison(self: bytes${length}, other: bytes${length}) -> bool {
+          self.value == other.value
+        }
+        fn distinct_comparison(self: bytes${length}, other: bytes${length}) -> bool {
+          self.value != other.value
+        }
+        fn greater_or_equal_comparison(self: bytes${length}, other: bytes${length}) -> bool {
+          self.value >= other.value
+        }
+        fn greater_comparison(self: bytes${length}, other: bytes${length}) -> bool {
+          self.value > other.value
+        }
+        // Bitwise 
+        fn and_bitwise(self: bytes${length}, other: bytes${length}) -> bytes${length} {
+          ${
+            max_width >= 128 // uN types lower than 128 bits don't have an implementation for bitwise operations
+              ? `bytes${length}{ value: self.value & other.value }`
+              : endent`
+              let self_value_felt252 = u${max_width}_to_felt252(self.value);
+              let other_value_felt252 = u${max_width}_to_felt252(other.value);
+              let self_value_u128 = u128_try_from_felt252(self_value_felt252).unwrap();
+              let other_value_u128 = u128_try_from_felt252(other_value_felt252).unwrap();
+              let result_u128 = self_value_u128 & other_value_u128;
+              let result_felt252 = u128_to_felt252(result_u128);
+              bytes${length}{ value: u${max_width}_try_from_felt252(result_felt252).unwrap() }
+            `
+          }
+        }
+        fn or_bitwise(self: bytes${length}, other: bytes${length}) -> bytes${length} {
+          ${
+            max_width >= 128 // uN types lower than 128 bits don't have an implementation for bitwise operations
+              ? `bytes${length}{ value: self.value | other.value }`
+              : endent`
+              let self_value_felt252 = u${max_width}_to_felt252(self.value);
+              let other_value_felt252 = u${max_width}_to_felt252(other.value);
+              let self_value_u128 = u128_try_from_felt252(self_value_felt252).unwrap();
+              let other_value_u128 = u128_try_from_felt252(other_value_felt252).unwrap();
+              let result_u128 = self_value_u128 | other_value_u128;
+              let result_felt252 = u128_to_felt252(result_u128);
+              bytes${length}{ value: u${max_width}_try_from_felt252(result_felt252).unwrap() }
+            `
```

Comment:
This code is the same for add, or and xor right? The only difference is the operator. I think you can write this body inside of a function that also receives the symbol of the operator and returns the string to be written in the cairo output